File: Assignment/app/src/main/java/com/assign/assignment/utils/AppConstants.java
package com.assign.assignment.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class AppConstants {
    public static final String SAVE_ACTIVITY_STATE = "activity";
    public static final String SAVE_FRAGMENT_STATE = "fragment";

    private boolean isNetworkAvailable(Context context) {
        ConnectivityManager connectivityManager
                = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
        return activeNetworkInfo != null && activeNetworkInfo.isConnected();
    }
    public static boolean isNetworkConnected(Context context) {
        ConnectivityManager cm = (ConnectivityManager)context.getSystemService(Context.CONNECTIVITY_SERVICE);

        return cm.getActiveNetworkInfo() != null;
    }

}

File: Assignment/app/src/main/java/com/assign/assignment/utils/AppConstants.kt
package com.assign.assignment.utils

import android.content.Context
import android.net.ConnectivityManager

object AppConstants {
    val SAVE_ACTIVITY_STATE = "activity"
    val SAVE_FRAGMENT_STATE = "fragment"

    fun isNetworkAvailable(context: Context): Boolean {
        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val activeNetworkInfo = connectivityManager.activeNetworkInfo
        return activeNetworkInfo != null && activeNetworkInfo.isConnected
    }
}
